src/cobald/composite/weighted.py

```diff
@@ -15,11 +16,10 @@ def demand(self):
     @demand.setter
     def demand(self, value):
         self._demand = value
-        total_supply = self.supply
         child_count = len(self.children)
         for pool in self.children:
             try:
-                pool.demand = value * pool.supply / total_supply
+                pool.demand = value * getattr(pool, self._weight) / self._total_weight
             except ZeroDivisionError:
                 pool.demand = value / child_count
```

Comment:
Is this still valid in context of configurable weight?

Reply:
Yes, that is still valid.